pkg/service/auth_context.go

```diff
@@ -3,54 +3,119 @@ package service
 import (
 	"fmt"
 	"strings"
+	"sync"
 
 	"github.com/3scale/authorino/pkg/config"
+	"github.com/3scale/authorino/pkg/config/internal"
 
-	"golang.org/x/net/context"
 	auth "github.com/envoyproxy/go-control-plane/envoy/service/auth/v2"
+	"golang.org/x/net/context"
 )
 
 // AuthContext holds the context of each auth request, including the request itself (sent by the client),
 // the auth config of the requested API and the lists of identity verifications, metadata add-ons and
 // authorization policies, and their corresponding results after evaluated
 type AuthContext struct {
 	ParentContext *context.Context
-	Request *auth.CheckRequest
-	API *APIConfig
+	Request       *auth.CheckRequest
+	API           *APIConfig
+
+	Identity      map[*config.IdentityConfig]interface{}
+	Metadata      map[*config.MetadataConfig]interface{}
+	Authorization map[*config.AuthorizationConfig]interface{}
```

Comment:
I would never use struct pointers as map keys. you can have may unexpected issues.

Instead, I would try to generate some kind of ID out of the structure. I would implement `ID` method in the struct and come up with some ID.

```
func (self *IdentityConfig) ID() string {
      // compute ID
}
```
```

Reply:
I'll comment this with the team, we might need to think further the context data struct.

Reply:
That's my fault. It was inherited from when I tried to rewrite in Golang as faithful as possible to what we had before in Ruby. In Ruby, the keys of the map (`Hash`) are the actual config objects. In Golang that wouldn't be possible, so I used the pointers.

...and BTW could you please give an example of when doing that would become an issue? Not that I am against improving it obviously; just wondering. E.g. can a declared variable move position in memory in Golang? Because that would be an example of this becoming a problem.